trunk/SUAVE/Methods/Performance/electric_V_h_diagram.py

```diff
@@ -0,0 +1,175 @@
+
+        Outputs:
+
+            climb_rate                      Climb Rates at Test Points  [ft/min]
+    """
+
+    # Unpack Inputs
+
+    g       = analyses.atmosphere.planet.sea_level_gravity
+    W       = vehicle.mass_properties.takeoff * g
+    S       = vehicle.reference_area
+
+    # Single Point Mission for Drag Determination
+
+    def mini_mission(altitude, speed):
+
+        mission = SUAVE.Analyses.Mission.Sequential_Segments()
+        mission.tag = 'the_mission'
+
+        segment = SUAVE.Analyses.Mission.Segments.Single_Point.Set_Speed_Set_Altitude_No_Propulsion()
+        segment.tag = 'single_point'
+        segment.analyses.extend(analyses)
+        segment.altitude = altitude
+        segment.air_speed = speed
+
+        mission.append_segment(segment)
+
+        return mission
+
+    # Specify Altitude and Speed Sample Points
+
+    alt_range       = np.linspace(0., altitude_ceiling, num=grid_points, endpoint=True)
+    speed_range     = np.linspace(0., max_speed, num = grid_points, endpoint=False)
+
+    # Initialize Climb Rate Grid
+
+    climb_rate = np.zeros((grid_points, grid_points))
+
+    # Loop Through Altitude and Speed Gridpoints
+
+    for alt_idx in range(grid_points):
+
+        altitude    = alt_range[alt_idx]
+        atmo_data   = analyses.atmosphere.compute_values(altitude, delta_isa)
+        rho         = atmo_data.density
+        Vs          = np.sqrt(2*W/(rho*S*1.4)) # Determine Vehicle Stall Speed
+
+        for speed_idx in range(grid_points):
+
+            V = speed_range[speed_idx]
+
+            if V > Vs: # Only Bother Calculating if Vehicle is Above Stall Speed
+
+                # Determine Vehicle Drag at Altitude and Speed
+
+                mission = mini_mission(altitude, V)
+                results = mission.evaluate()
+
+                D = -results.segments.single_point.conditions.frames.wind.drag_force_vector[0][0]
+
+                # Determine Propeller Power at Altitude and Speed
+
+                P = propeller_single_point(vehicle.propulsors.battery_propeller,
+                                           analyses,
+                                           pitch=0.,
+                                           omega=test_omega,
+                                           altitude=altitude,
+                                           delta_isa=0.,
+                                           speed=V).power
+
+                # Check if Propeller Power Exceeds Max Battery Power, Switch to Max Battery Power if So
+
+                P = np.min([P, vehicle.propulsors.battery_propeller.battery.max_power])
+
+                # Determine Climb Rate (ref. Raymer)
+
+                cr = 1/W * (P - D*V)
+
+                # If Climb Rate is Negative, Replace with 0 for Easy Contour-Finding
+
+                climb_rate[speed_idx, alt_idx] = np.max([0., cr])
+
+
+    climb_rate  = climb_rate / Units['ft/min']
+
+    if display_plot:
```

Comment:
in your regression make display_plot true. It will be suppressed so it won't show but still run those lines of code and add to the coverage.

Reply:
Done